File: backend/src/main/java/com/laptopdb/backend/entity/Display.java
package com.laptopdb.backend.entity;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
@Entity
@Table(name = "displays")
public class Display {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull(message = "Display size cannot be null")
    @Column(name = "size_inch", nullable = false, precision = 3, scale = 1)
    private BigDecimal sizeInch;

    @NotBlank(message = "Resolution cannot be blank")
    @Column(nullable = false, length = 20)
    private String resolution;

    @NotNull(message = "Refresh rate cannot be null")
    @Column(name = "refresh_rate_hz", nullable = false)
    private Integer refreshRateHz;

    @NotBlank(message = "Panel type cannot be blank")
    @Column(name = "panel_type", nullable = false, length = 20)
    private String panelType;

    @Column(name = "brightness_nits")
    private Integer brightnessNits;
}

File: frontend/lib/models/display.dart
class Display {
  final int id;
  final double sizeInch;
  final String resolution;
  final int refreshRateHz;
  final String panelType;
  final int? brightnessNits;

  Display({
    required this.id,
    required this.sizeInch,
    required this.resolution,
    required this.refreshRateHz,
    required this.panelType,
    this.brightnessNits,
  });

  factory Display.fromJson(Map<String, dynamic> json) {
    return Display(
      id: json['id'] ?? 0,
      sizeInch: (json['sizeInch'] ?? 0).toDouble(),
      resolution: json['resolution'] ?? '',
      refreshRateHz: json['refreshRateHz'] ?? 0,
      panelType: json['panelType'] ?? '',
      brightnessNits: json['brightnessNits'],
    );
  }

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{
      'sizeInch': sizeInch,
      'resolution': resolution,
      'refreshRateHz': refreshRateHz,
      'panelType': panelType,
    };
    if (brightnessNits != null) map['brightnessNits'] = brightnessNits;
    return map;
  }

  String get displayLabel =>
      '${sizeInch}" $resolution ${refreshRateHz}Hz $panelType';
}
